$NetBSD$

Split configuration between -client and -server.
Do not run libmysql_api_test, requires LD_LIBRARY_PATH.

--- libmysql/CMakeLists.txt.orig	2022-02-28 19:55:41.000000000 +0000
+++ libmysql/CMakeLists.txt
@@ -272,6 +272,7 @@ IF(WIN32)
 ENDIF()
 
 # LDAP authentication SASL client plug-in
+IF(NOT WITHOUT_SERVER)
 ADD_SUBDIRECTORY(authentication_ldap)
 
 # FIDO authentication client plugin
@@ -282,13 +283,20 @@ ADD_SUBDIRECTORY(authentication_kerberos
 
 # authentication IAM client plug-in
 ADD_SUBDIRECTORY(authentication_oci_client)
+ENDIF()
 
 # Merge several convenience libraries into one big perconaserverclient
+IF(WITHOUT_SERVER)
 MERGE_CONVENIENCE_LIBRARIES(perconaserverclient ${LIBS_TO_MERGE}
   COMPONENT Development
   LINK_LIBRARIES ${LIBS_TO_LINK}
   )
-
+ELSE()
+MERGE_CONVENIENCE_LIBRARIES(perconaserverclient SKIP_INSTALL ${LIBS_TO_MERGE}
+  COMPONENT Development
+  LINK_LIBRARIES ${LIBS_TO_LINK}
+  )
+ENDIF() 
 # Visual Studio users need debug  static library for debug projects
 IF(MSVC)
   INSTALL_DEBUG_TARGET(perconaserverclient DESTINATION ${INSTALL_LIBDIR}/debug)
@@ -334,6 +342,7 @@ ENDIF()
 
 # Merge several convenience libraries into one big perconaserverclient
 # and link them together into shared library.
+IF(WITHOUT_SERVER)
 MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
   EXPORTS
   ${CLIENT_API_FUNCTIONS}
@@ -345,6 +354,19 @@ MERGE_LIBRARIES_SHARED(libmysql ${LIBS_T
   ${UNIX_SOVERSION}
   ${UNIX_VERSION}
   )
+ELSE()
+MERGE_LIBRARIES_SHARED(libmysql SKIP_INSTALL ${LIBS_TO_MERGE}
+  EXPORTS
+  ${CLIENT_API_FUNCTIONS}
+  ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+  ${CLIENT_API_NONBLOCKING_FUNCTIONS}
+  COMPONENT SharedLibraries
+  LINK_LIBRARIES ${LIBS_TO_LINK}
+  ${UNIX_OUTPUT_NAME}
+  ${UNIX_SOVERSION}
+  ${UNIX_VERSION}
+  )
+ENDIF()
 
 # Downgrade warning for strncat in my_crypt_genhash.
 IF((WITH_LTO OR CMAKE_COMPILER_FLAG_WITH_LTO) AND MY_COMPILER_IS_GNU)
@@ -400,6 +422,7 @@ CONFIGURE_FILE(api_test.c.in ${CMAKE_CUR
 # from @CLIENT_API_FUNCTIONS@ are declared by <mysql.h>. It will fail
 # to run if not all of these symbols are exported by the library.
 #
+IF(NOT WITHOUT_SERVER)
 MYSQL_ADD_EXECUTABLE(libmysql_api_test
   ${CMAKE_CURRENT_BINARY_DIR}/api_test.c
   LINK_LIBRARIES libmysql ${LIBRT}
@@ -422,9 +445,12 @@ IF(HAS_WARN_FLAG)
     COMPILE_FLAGS ${HAS_WARN_FLAG}
     )
 ENDIF()
+ENDIF()
 
 # Verify that libmysql_api_test runs OK
+IF(0)
 ADD_CUSTOM_COMMAND(TARGET libmysql_api_test POST_BUILD
   COMMAND libmysql_api_test
   > ${CMAKE_CURRENT_BINARY_DIR}/libmysql_api_test.out
   )
+ENDIF()
